jsonalchemy/json_encoder.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of JSONAlchemy.
+# Copyright (C) 2015 CERN.
+#
+# JSONAlchemy is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# JSONAlchemy is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with JSONAlchemy; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""JSONAlchemy aware JSON encoder."""
+
+from jsonalchemy.custom_encoder import CustomEncoder
+from jsonalchemy.wrappers import JSONArray
+from jsonalchemy.wrappers import JSONObject
+
+from six import iteritems
+
+
+class JSONAlchemyEncoder(CustomEncoder):
+    def default(self, obj):
+        if isinstance(obj, JSONObject):
+            result = {k: v for (k, v) in iteritems(obj)}
+            for el in obj.schema['properties']:
+                if ('getter' in obj.schema['properties'][el] or
+                        'template' in obj.schema['properties'][el]):
+                    result[el] = obj[el]
+
+            return result
+        elif isinstance(obj, JSONArray):
+            return [self.default(el) for el in obj]
+
```

Comment:
Maybe here we should fallback to `JSONEncoder`'s default, and throw a type error.